File: problems/67/Main.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>
#include <cstring>
#include <string>
#include <cctype>
#include <stack>
#include <queue>
#include <list>
#include <vector>
#include <map>
#include <sstream>
#include <cmath>
#include <bitset>
#include <utility>
#include <set>
#include <numeric>
#include <ctime>
#include <math.h>

using namespace std;

class Solution {
	public:
		int gcd(int a, int b);
};

//--SPLIT--
//-int Solution::gcd(int a, int b) {
//-}

//--SPLIT--
int Solution::gcd(int a, int b) {
	if (a == 0) {
		return b;
	}
	return gcd(b%a, a);
}

//--SPLIT--

int main() {
	int testcases;
	cin >> testcases;
    Solution s;
	for (int t = 0; t < testcases; t++) {
		int a, b; cin >> a >> b;
		cout << s.gcd(a, b) << "\n";
	}
	return 0;
}

File: problems/67/Main.py
import math

#--SPLIT--
#-class Solution:
    # @param a : integer
    # @param b : integer
    # @return an integer
#-    def gcd(self, a, b):

#--SPLIT--
class Solution:
    def gcd(self, a, b):
        if a  == 0:
            return b
        return self.gcd(b%a, a)


#--SPLIT--

if __name__ == "__main__":
    test_cases = int(raw_input())
    for i in range(test_cases):
        a, b = [int(l) for l in raw_input().split(" ")]
        s = Solution()
        print(s.gcd(a, b))